Makefile

```diff
@@ -18,16 +18,29 @@ ifneq ($(MONO_OPTIONS),)
 export MONO_OPTIONS
 endif
 
-include build-tools/scripts/msbuild.mk
 
+include build-tools/scripts/msbuild.mk
 all::
 	$(MSBUILD) $(MSBUILD_FLAGS) $(SOLUTION)
 
 all-tests::
 	MSBUILD="$(MSBUILD)" tools/scripts/xabuild $(MSBUILD_FLAGS) Xamarin.Android-Tests.sln
 
-prepare:: prepare-msbuild
 
+include build-tools/scripts/Dependencies.mk
+ifneq ($(OS),Linux)
+export LINUX_DISTRO=none
+endif
+prepare:: linux-prepare-$(LINUX_DISTRO) prepare-msbuild
```

Comment:
This seems unwise; won't this run `linux-prepare-none` for *all* operating systems, including macOS? Similarly lines 36-43 and the use of `$(BINFMT_MISC_TROUBLE)` and `/proc/sys/fs/binfmt_misc`: these uses will fail on macOS, and I see nothing that would *prevent* execution on macOS.